python/tvm/relay/frontend/tensorflow.py

```diff
@@ -3682,6 +3729,30 @@ def from_tensorflow(graph, layout="NHWC", shape=None, outputs=None):
     params : dict of str to tvm.nd.NDArray
         Dict of converted parameters stored in tvm.nd.NDArray format
     """
+    from tensorflow.python.eager.function import ConcreteFunction
+
+    if isinstance(graph, ConcreteFunction):
```

Comment:
would it be better if we create a wrapper over from_tensorflow_concrete_function() instead of overloading the parameter "graph" to be either GraphDef or ConcreteFunction? 

Another question: Is ConcreteFunction the right entry point? Why not start from the saved_model format? When TVM is used to compile a model for inference, it starts from a saved_model. This way there is no need to use _build_signature_def in order to optimize the graph. Maybe _build_signature_def can be part of the test utility code.

Reply:
Wrapper will change frontend API. We could seek community opinion on this.

I thought of saved_model, but we already have TFParser (python/tvm/relay/frontend/tensorflow_parser.py) helper that handles saved_model formats. I think we could make ```from_tensorflow``` being single entry for GraphDef / ConcreteFunction / Saved Model Dir.  TFParser can handle all transforms until it results GraphDef.

TFHub URL to GraphDef can also be added to TFParser, but hub.KerasLayer has bunch or args to handle some times. We could hold it for now.

I will amend this PR with these changes and also TF1.x / TF2.x compatibility we discussed.